pkg/pool/connection_pool.go

```diff
@@ -10,13 +10,16 @@ import (
 
 // ConnectionPool is the in memory implementation of a connection Pool
 type ConnectionPool struct {
-	mu    sync.RWMutex
+	mu    *sync.RWMutex
```

Comment:
There's no need for this to be a pointer.

Reply:
changed